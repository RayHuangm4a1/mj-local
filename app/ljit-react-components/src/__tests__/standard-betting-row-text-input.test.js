import React from 'react';
import { shallow, mount, } from 'enzyme';
import TextInput from '../components/standard-betting-row/text-input';

describe('TextInput', () => {
	it('should be selectable by class ljit-text-input', () => {
		const wrapper = shallow(<TextInput/>);

		expect(wrapper.hasClass('ljit-text-input')).toEqual(true);
	});
	it('should handle default props', () => {
		const {
			data,
			placeholder,
			onChangeText,
		} = TextInput.defaultProps;

		expect(data).toEqual('');
		expect(placeholder).toEqual('');
		expect(onChangeText).toBeInstanceOf(Function);
	});
	it('should renders correctly', () => {
		const data = 'mock-data';
		const placeholder = 'mock-placeholder';
		const onChangeText = () => {};
		const wrapper = shallow(
			<TextInput 
				data={data}
				placeholder={placeholder}
				onChangeText={onChangeText}
			/>
		);

		expect(wrapper).toMatchSnapshot();
	});
	it('should mount in a full DOM', () => {
		const data = 'mock-data';
		const placeholder = 'mock-placeholder';
		const onChangeText = () => {};
		const wrapper = mount(
			<TextInput 
				data={data}
				placeholder={placeholder}
				onChangeText={onChangeText}
			/>
		);

		expect(wrapper.props().data).toBe(data);
		expect(wrapper.props().placeholder).toBe(placeholder);
		expect(wrapper.props().onChangeText).toEqual(onChangeText);
	});
});
