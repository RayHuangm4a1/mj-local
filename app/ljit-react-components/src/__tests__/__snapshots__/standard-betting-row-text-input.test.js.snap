// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput should renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StandardBettingRowTextInput
    data="mock-data"
    onChangeText={[Function]}
    placeholder="mock-placeholder"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="ljit-text-input__button-group"
        >
          <Button
            className="ljit-text-input__button"
            color=""
            disabled={false}
            onClick={[Function]}
            outline="solid"
          >
             导入注单 
          </Button>
          <Button
            className="ljit-text-input__button"
            color=""
            disabled={false}
            onClick={[Function]}
            outline="solid"
          >
             删除重复号 
          </Button>
          <Button
            className="ljit-text-input__button"
            color=""
            disabled={false}
            onClick={[Function]}
            outline="solid"
          >
             清 空 
          </Button>
        </div>,
        <InputTextarea
          disabled={false}
          maxRows={7}
          minRows={7}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onPressEnter={[Function]}
          placeholder="mock-placeholder"
          readOnly={false}
          value="mock-data"
        />,
      ],
      "className": "ljit-text-input",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Button
              className="ljit-text-input__button"
              color=""
              disabled={false}
              onClick={[Function]}
              outline="solid"
            >
               导入注单 
            </Button>,
            <Button
              className="ljit-text-input__button"
              color=""
              disabled={false}
              onClick={[Function]}
              outline="solid"
            >
               删除重复号 
            </Button>,
            <Button
              className="ljit-text-input__button"
              color=""
              disabled={false}
              onClick={[Function]}
              outline="solid"
            >
               清 空 
            </Button>,
          ],
          "className": "ljit-text-input__button-group",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": " 导入注单 ",
              "className": "ljit-text-input__button",
              "color": "",
              "disabled": false,
              "onClick": [Function],
              "outline": "solid",
            },
            "ref": null,
            "rendered": " 导入注单 ",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": " 删除重复号 ",
              "className": "ljit-text-input__button",
              "color": "",
              "disabled": false,
              "onClick": [Function],
              "outline": "solid",
            },
            "ref": null,
            "rendered": " 删除重复号 ",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": " 清 空 ",
              "className": "ljit-text-input__button",
              "color": "",
              "disabled": false,
              "onClick": [Function],
              "outline": "solid",
            },
            "ref": null,
            "rendered": " 清 空 ",
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "disabled": false,
          "maxRows": 7,
          "minRows": 7,
          "onBlur": [Function],
          "onChange": [Function],
          "onFocus": [Function],
          "onPressEnter": [Function],
          "placeholder": "mock-placeholder",
          "readOnly": false,
          "value": "mock-data",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="ljit-text-input__button-group"
          >
            <Button
              className="ljit-text-input__button"
              color=""
              disabled={false}
              onClick={[Function]}
              outline="solid"
            >
               导入注单 
            </Button>
            <Button
              className="ljit-text-input__button"
              color=""
              disabled={false}
              onClick={[Function]}
              outline="solid"
            >
               删除重复号 
            </Button>
            <Button
              className="ljit-text-input__button"
              color=""
              disabled={false}
              onClick={[Function]}
              outline="solid"
            >
               清 空 
            </Button>
          </div>,
          <InputTextarea
            disabled={false}
            maxRows={7}
            minRows={7}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onPressEnter={[Function]}
            placeholder="mock-placeholder"
            readOnly={false}
            value="mock-data"
          />,
        ],
        "className": "ljit-text-input",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Button
                className="ljit-text-input__button"
                color=""
                disabled={false}
                onClick={[Function]}
                outline="solid"
              >
                 导入注单 
              </Button>,
              <Button
                className="ljit-text-input__button"
                color=""
                disabled={false}
                onClick={[Function]}
                outline="solid"
              >
                 删除重复号 
              </Button>,
              <Button
                className="ljit-text-input__button"
                color=""
                disabled={false}
                onClick={[Function]}
                outline="solid"
              >
                 清 空 
              </Button>,
            ],
            "className": "ljit-text-input__button-group",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": " 导入注单 ",
                "className": "ljit-text-input__button",
                "color": "",
                "disabled": false,
                "onClick": [Function],
                "outline": "solid",
              },
              "ref": null,
              "rendered": " 导入注单 ",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": " 删除重复号 ",
                "className": "ljit-text-input__button",
                "color": "",
                "disabled": false,
                "onClick": [Function],
                "outline": "solid",
              },
              "ref": null,
              "rendered": " 删除重复号 ",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": " 清 空 ",
                "className": "ljit-text-input__button",
                "color": "",
                "disabled": false,
                "onClick": [Function],
                "outline": "solid",
              },
              "ref": null,
              "rendered": " 清 空 ",
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "disabled": false,
            "maxRows": 7,
            "minRows": 7,
            "onBlur": [Function],
            "onChange": [Function],
            "onFocus": [Function],
            "onPressEnter": [Function],
            "placeholder": "mock-placeholder",
            "readOnly": false,
            "value": "mock-data",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
